src/app/container/register-tool/register-tool.service.ts

```diff
@@ -109,6 +110,30 @@ export class RegisterToolService {
         );
     }
 
+    /**
+     * Registers a hosted tool
+     * @param  hostedTool hosted tool object
+     */
+    registerHostedTool(hostedTool): void {
+      const splitPath = hostedTool.path.split('/');
+      const namespace = splitPath[0];
+      const name = splitPath[1];
+      this.stateService.setRefreshMessage('Registering tool...');
+      this.hostedService.createHostedTool(
+          name,
+          'cwl',
+          hostedTool.registry,
+          namespace).subscribe(result => {
```

Comment:
type

Reply:
where?

Reply:
Where?

Reply:
On the result? I can't put Tool as a type there.